geoh5py/shared/merging/base.py

```diff
@@ -104,10 +109,14 @@ def merge_data(cls, out_entity, input_entities: list[ObjectBase]):
                     data_dict[label].values = values
 
             data_count["VERTEX"] += (
-                input_entity.n_vertices if input_entity.n_vertices is not None else 0
+                input_entity.n_vertices + cls._ghost
```

Comment:
I would vote that we keep the ghost attribute to the DrapeMerger only, as it is only used specifically there. Consider simply re-implementing merge_data under the class

```
def merge_data(cls, out_entity, input_entities: list[ObjectBase]):
   - call super().merge_data(out_entity, input_entities)
   - Loop over the children and shift the values to account for the ghost cells
```

Reply:
That's what I did in the first place.
.
I agree it's not useful in BaseMerger, but it's just a tiny +2 somewhere. 

option 1) It seems to me that calling super() and rerunning a second loop to insert nan values between the previous values is not efficient as it use a second loop and also introduces a lot of code. 

option 2) Just copying the full function and adding +2  introduces a lot of duplicated code too. 

option 3) But having a class attribute just added 4 lines of code in total, without changing the performances...


I'm letting you choose the option in the comment and correct the code the way you want.

Reply:
I am not too worried about the efficiency part, since the number of child will be small, and re-assigning values is pretty fast. I would rather the base class to be simple and not contain special cases

Reply:
Added a new loop using super to redefined merge_data() in DrapeModel